js/repl.ts

```diff
@@ -113,42 +156,23 @@ function evaluate(code: string): void {
       console.error("Thrown:", errInfo.thrown);
     }
   }
+  return true;
 }
 
-async function readBlock(
-  rid: number,
-  prompt: string,
-  continuedPrompt: string
-): Promise<string> {
-  let code = "";
-  do {
-    code += await readline(rid, prompt);
-    prompt = continuedPrompt;
-  } while (parenthesesAreOpen(code));
-  return code;
-}
-
-// modified from
-// https://codereview.stackexchange.com/a/46039/148556
-function parenthesesAreOpen(code: string): boolean {
-  const parentheses = "[]{}()";
-  const stack = [];
-
-  for (const ch of code) {
-    const bracePosition = parentheses.indexOf(ch);
-
-    if (bracePosition === -1) {
-      // not a paren
-      continue;
-    }
-
-    if (bracePosition % 2 === 0) {
-      stack.push(bracePosition + 1); // push next expected brace position
-    } else {
-      if (stack.pop() !== bracePosition) {
-        return false;
-      }
-    }
-  }
-  return stack.length > 0;
+// Error messages that allow users to continue input
+// instead of throwing an error to REPL
+// ref: https://github.com/v8/v8/blob/master/src/message-template.h
+// TODO(kevinkassimo): this list might not be comprehensive
+const recoverableErrorMessages = [
+  "Unexpected end of input", // { or [ or (
+  "Missing initializer in const declaration", // const a
+  "Missing catch or finally after try", // try {}
+  "missing ) after argument list", // console.log(1
+  "Unterminated template literal" // `template
+  // TODO(kevinkassimo): need a parser to handling errors such as:
+  // "Missing } in template expression" // `${ or `${ a 123 }`
+];
```

Comment:
Hm. This isn't the most elegant solution... That said, if it makes life easier for REPL users, I think it's sufficiently self contained. Ultimately I think we should use a proper JS parser - we have two in the deno.exe already.  I don't have a good suggestion for how to access those parsers from here at the moment. We can revisit later.